### DISCUSSION TITLE
enumerating RoleAssigments and RoleDefinitions of all items

### DISCUSSION THREAD

**Discussion by EmkioA** [pnp/pnpcore#1309] [Q&A]
Hello, i have a small code enumerating roleassignements, and i want to know for who the roledefinitions are assigned to...  Into the [iRoleAssignments ](https://pnp.github.io/pnpcore/api/PnP.Core.Model.Security.IRoleAssignment.html) we have access to the PrincipaleId but how can i know if this id is a user or a group?  

Example : i receive a PrincipaleId = 3 and if i lookinto the web.GetUserById(3) i receive an error (not found). and if i look on the site and i click on the groupe name, i am redirect to /_layouts/15/people.aspx?MembershipGroupId=3

How can i distinct Users from Groups into the IRoleAssigments?

**Comment by jansenbe** [comment]
@EmkioA : probably the fastest solution is to lookup the SharePoint groups upfront and then check whether the principal id matches with one of the group id's.

**Reply by EmkioA** [reply]
@jansenbe  Thank you for the answer,

I have think of it, but what i do if both id matches? if y have a groupid 3 and a userid 3 too?  wich one i choose?

There is a [PrincipalType](https://pnp.github.io/pnpcore/api/PnP.Core.Model.Security.PrincipalType.html) enum but i cant find any link between the IRoleAssigment, IRoleDefinition and PrincipalType  i am missing something?

**Reply by jansenbe** [reply]
PrincipalId should be unique

**Reply by EmkioA** [reply]
what make me unsure is they are into 2 separates collection 
iWeb.SiteGroups and iWeb.SiteUsers

**Reply by jansenbe** [reply]
They are unique
